platform/admin/admin-configuration-configinstaller/src/main/java/org/codice/ddf/admin/configuration/ConfigurationInstaller.java

```diff
@@ -0,0 +1,201 @@
+import org.slf4j.LoggerFactory;
+
+/**
+ * Addresses shortcomings of the Felix {@code ConfigInstaller} with our own implementation that properly
+ * syncs the etc directory using the {@code felix.fileinstall.filename} property. Handling a config delete
+ * is fairly straight forward. The following outcomes are possible when an update occurs:
+ * <p>
+ * <table>
+ * <tr>
+ * <td>felixFileOriginal</td><td>felixFileNew</td><td>comments</td>
+ * </tr>
+ * <tr>
+ * <td>null</td><td>null</td><td>Felix file prop wasn't tracked and is still not tracked</td>
+ * </tr>
+ * <tr>
+ * <td>null</td><td>X</td><td>New config with a new Felix file prop to track</td>
+ * </tr>
+ * <tr>
+ * <td>Y</td><td>null</td><td>Felix file prop no longer being tracked, delete config file and remove from map</td>
+ * </tr>
+ * <tr>
+ * <td>Y</td><td>X</td><td>Felix file prop updated, delete config file and update map</td>
+ * </tr>
+ * <tr>
+ * <td>Y</td><td>Y</td><td>Felix file prop did not change, do nothing</td>
+ * </tr>
+ * </table>
+ *
+ * @see #handleUpdate(FelixConfig)
+ * @see #handleDelete(String)
+ */
+public class ConfigurationInstaller implements SynchronousConfigurationListener {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ConfigurationInstaller.class);
+
+    private final ConfigurationAdmin configurationAdmin;
+
+    private final ConfiguratorFactory configuratorFactory;
+
+    private final ConfigActions configActions;
+
+    private Map<String, File> pidFileMap;
+
+    public ConfigurationInstaller(ConfigurationAdmin configurationAdmin,
+            ConfiguratorFactory configuratorFactory, ConfigActions configActions) {
+        this.configurationAdmin = configurationAdmin;
+        this.pidFileMap = Collections.emptyMap();
+
+        this.configuratorFactory = configuratorFactory;
+        this.configActions = configActions;
+    }
+
+    /**
+     * @return a read-only map of configuration pids to felix file install properties.
+     */
+    public Map<String, File> getPidFileMap() {
+        return Collections.unmodifiableMap(pidFileMap);
+    }
+
+    public void init() throws IOException, InvalidSyntaxException {
+        Configuration[] configs = configurationAdmin.listConfigurations(null);
+        if (configs == null) {
+            return;
+        }
+        this.pidFileMap = Arrays.stream(configs)
+                .map(FelixConfig::new)
+                .filter(config -> config.getFile() != null)
+                .collect(Collectors.toMap(FelixConfig::getPid, FelixConfig::getFile));
+    }
+
+    @Override
+    public void configurationEvent(ConfigurationEvent configurationEvent) {
+        switch (configurationEvent.getType()) {
+        case CM_UPDATED:
+            LOGGER.debug("Handling update");
+            Configuration configuration = getConfiguration(configurationEvent);
+            if (configuration == null) {
+                break;
+            }
+            handleUpdate(new FelixConfig(configuration));
```

Comment:
✏️ I'm not sure I see the advantage of pulling the logic to get the configuration up to here instead of just passing the `configurationEvent` down to the `handleUpdate` method. The same applies to your private `handleDelete` method and the necessity of extracting the pid in this method.